src/iv/ivmain.cpp

```diff
@@ -108,8 +108,38 @@ main(int argc, char* argv[])
     mainWin->activateWindow();
 
     // Add the images
-    for (auto& f : ap["filename"].as_vec<std::string>())
-        mainWin->add_image(f);
+    for (auto& f : ap["filename"].as_vec<std::string>()) {
+        // Check if the file exists
+        if (!Filesystem::exists(f)) {
+            std::cerr << "Error: File or directory does not exist: " << f << "\n";
+            continue;
+        }
+
+        if (Filesystem::is_directory(f)) {
+            // If f is a directory, iterate through its files
+            std::vector<std::string> files;
+            Filesystem::get_directory_entries(f, files);
+
+            std::vector<std::string> validImages;  // Vector to hold valid images
+            for (auto& file : files) {
+                auto in = ImageInput::open(file);
+                if (in) {
+                    validImages.push_back(file);
+                    in->close();  // Close the ImageInput object to free resources
+                }
+            }
```

Comment:
I think that this is technically correct (except for the fact that if the open fails, you are not retrieving any error messages to throw them out).

But I think this could be unnecessarily expensive. What you need to know is that what ImageInput::open does is first try the image reader corresponding to the file's extension (if there is one), but if that fails, then it will try every other file reader, just in case the file is mis-named. So if the directory is filled with lots of non-image files, it might try to open it with each of a dozen or more reader types, all to no avail.

So one thing I would suggest is that you only open() the files that match one of the known extensions corresponding to a supported image file. (See this](https://openimageio.readthedocs.io/en/v2.5.4.0/imageioapi.html#_CPPv4N4OIIO12getattributeE11string_view8TypeDescPv) for the getattribute query that will give you all known extensions).

Also, since iv is going to do everything through the ImageCache, it may be good to use it here, too. Assuming that you have set up

    ImageCache* cache = ImageCache::create();
    ustring uexists("exists");

outside the loop, then you can check if a file is a valid image in one step:

    int exists = 0;
    bool ok = cache->get_image_info(ustring(file), 0, 0, uexists, OIIO::TypeInt, &exists);
    if (ok && exists)
        validImages.push_back(file);

rather than the open call. I'm not certain that this is significantly better, but it may be, because it will prevent you from having to open each file twice (remember, you're probably gonna want to display each image, too).